src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -848,6 +845,24 @@ abstract class BTypes {
    * infos using `get`, but it reports inliner warnings for missing infos that prevent inlining.
    */
   final case class ClassBType(internalName: InternalName) extends RefBType {
+
+    override def descriptor: String = {
+      if (descriptorCache eq null) {
+        val sb = new java.lang.StringBuilder(internalName.length + 2)
+        buildDescriptor(sb)
```

Comment:
there will probably be concurrent accesses to this method once we parallelize the backend, no?